src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_sdfg.py

```diff
@@ -160,36 +158,62 @@ def _make_array_shape_and_strides(
+        transient: bool = True,
+        is_tuple_member: bool = False,
+    ) -> list[tuple[str, ts.DataType]]:
         """
         Add storage for data containers used in the SDFG. For fields, it allocates dace arrays,
         while scalars are stored as SDFG symbols.
 
         The fields used as temporary arrays, when `transient = True`, are allocated and exist
         only within the SDFG; when `transient = False`, the fields have to be allocated outside
         and have to be passed as array arguments to the SDFG.
+
+        Returns:
+            List of data containers or symbols allocated as storage. This is a list, not a single value,
+            because in case of tuples we flat the tuple fields (eventually nested) and allocate storage
+            for each tuple element.
         """
-        if isinstance(symbol_type, ts.FieldType):
+        if isinstance(symbol_type, ts.TupleType):
+            tuple_fields = []
+            for tname, tsymbol_type in dace_gtir_utils.get_tuple_fields(
+                name, symbol_type, flatten=True
+            ):
+                tuple_fields.extend(
+                    self._add_storage(sdfg, tname, tsymbol_type, transient, is_tuple_member=True)
+                )
+            return tuple_fields
+
+        elif isinstance(symbol_type, ts.FieldType):
             dtype = dace_utils.as_dace_type(symbol_type.dtype)
             # use symbolic shape, which allows to invoke the program with fields of different size;
             # and symbolic strides, which enables decoupling the memory layout from generated code.
             sym_shape, sym_strides = self._make_array_shape_and_strides(name, symbol_type.dims)
             sdfg.add_array(name, sym_shape, dtype, strides=sym_strides, transient=transient)
+
         elif isinstance(symbol_type, ts.ScalarType):
-            assert isinstance(symbol_type, ts.ScalarType)
             dtype = dace_utils.as_dace_type(symbol_type)
-            # Scalar arguments passed to the program are represented as symbols in DaCe SDFG.
+            # Scalar arguments passed to the program are represented as symbols in DaCe SDFG;
+            # the exception are members of tuple arguments, that are represented as scalar containers.
             # The field size is sometimes passed as scalar argument to the program, so we have to
             # check if the shape symbol was already allocated by `_make_array_shape_and_strides`.
             # We assume that the scalar argument for field size always follows the field argument.
-            if name in sdfg.symbols:
+            if is_tuple_member:
+                sdfg.add_scalar(name, dtype, transient=transient)
+            elif name in sdfg.symbols:
                 assert sdfg.symbols[name].dtype == dtype
             else:
                 sdfg.add_symbol(name, dtype)
+
         else:
             raise RuntimeError(f"Data type '{type(symbol_type)}' not supported.")
 
+        return [(name, symbol_type)]
```

Comment:
I would either early return everywhere or not at all, otherwise it's just harder to follow the control flow.